src/featureFlags/flags/baseEnabled.ts

```diff
@@ -0,0 +1,9 @@
+import { BaseVariant, FeatureFlag, useBaseFlag } from '../index'
+
+export function useBaseEnabledFlag(): BaseVariant {
+  return useBaseFlag(FeatureFlag.baseEnabled)
+}
+
+export function useBaseEnabled(): boolean {
+  return useBaseEnabledFlag() === BaseVariant.Enabled
+}
```

Comment:
nit: add the below helper function and can be used in several locations where you need the mapping to reduce repetition
```suggestion
}

export function useBaseEnabledChains(): Record<number, boolean> {
  const baseEnabled = useBaseEnabled()
  return {
    [ChainId.BASE]: baseEnabled,
    [ChainId.BASE_GOERLI]: baseEnabled,
  }
}
```